notification.c

```diff
@@ -157,25 +209,20 @@ size_t format_notification(struct mako_notification *notif, const char *format,
 		}
 		len += chunk_len;
 
-		const char *value = NULL;
+		char *value = NULL;
 		bool markup = false;
-		switch (current[1]) {
-		case '%':
+
+		if (current[1] == '%') { 
 			value = "%";
-			break;
-		case 'a':
-			value = notif->app_name;
-			break;
-		case 's':
-			value = notif->summary;
-			break;
-		case 'b':
-			value = notif->body;
-			markup = true;
-			break;
+		} else {
+			value =	format_func(current[1], data);
 		}
 		if (value == NULL) {
-			value = "";
+			value = strdup("");
+		}
+
+		if (current[1] == 'b') {
```

Comment:
I'm not sure how to handle setting `markup` yet. You currently had it just for handling the body, so I just included it here like this for now, but it should probably be handled differently.

Reply:
Basically we want to interpret markup (ie. tags like `<b>`) only in the body of the notification. If the notification body is `<b>Hi there</b>`, we want to print "Hi there" in bold. If the notification summary is `<b>Hi there</b>`, we want to print "<b>Hi there</b>" as is.

Maybe `&markup` can be passed as an argument to `format_func`.